taskvine/src/manager/vine_manager.c

```diff
@@ -1662,6 +1665,23 @@ static vine_result_code_t get_result(struct vine_manager *q, struct vine_worker_
 	/* Consume the stdout data immediately following this message. */
 	get_stdout(q, w, t, output_length);
 
+	/* If a library task fails, remove this instance  */
+	/* handle failed library instances */
+	if (task_status == VINE_RESULT_LIBRARY_FAILED) {
+		debug(D_VINE, "Task %d library %s failed", t->task_id, t->provides_library);
+		reset_task_to_state(q, t, VINE_TASK_RETRIEVED);
```

Comment:
Just do the things specific to libraries here, such as counting up the failures and emitting a message.

But don't shortcut the task transitions: we want all tasks to go through the same transitions for all sorts of reasons: accounting, dealing with output files, etc.  Instead of changing the state here, just fall through the code below and do the normal itable_remove and change_task_state.

Reply:
In fact, don't even bother checking for VINE_RESULT_LIBRARY_FAILED b/c a library that exits "normally" is still failing from our perspective.